docs/src/main/asciidoc/security-csrf-prevention.adoc

```diff
@@ -0,0 +1,200 @@
+
+Quarkus Security provides a CSRF prevention feature which consists of a xref:resteasy-reactive.adoc[Resteasy Reactive] server filter which creates and verifies CSRF tokens and an HTML form parameter provider which supports the xref:qute-reference.adoc#injecting-beans-directly-in-templates[injection of CSRF tokens in Qute templates].
+
+== Creating the Project
+
+First, we need a new project.
+Create a new project with the following command:
+
+:create-app-artifact-id: security-csrf-prevention
+:create-app-extensions: csrf-reactive
+include::{includes}/devtools/create-app.adoc[]
+
+This command generates a project which imports the `csrf-reactive` extension.
+
+If you already have your Quarkus project configured, you can add the `csrf-reactive` extension
+to your project by running the following command in your project base directory:
+
+:add-extension-extensions: csrf-reactive
+include::{includes}/devtools/extension-add.adoc[]
+
+This will add the following to your build file:
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-csrf-reactive</artifactId>
+</dependency>
+----
+
+[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
+.build.gradle
+----
+implementation("io.quarkus:quarkus-csrf-reactive")
+----
+
+Next lets add a Qute template producing an HTML form:
+
+[source,html]
+----
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="UTF-8">
+<title>User Name Input</title> 
+</head>
+<body>
+    <h1>User Name Input</h1>
+
+    <form action="/service/csrfTokenForm" method="post">
+    	<input type="hidden" name="{inject:csrf.parameterName}" value="{inject:csrf.token}" />  <1>
+    	
+    	<p>Your Name: <input type="text" name="name" /></p>
+    	<p><input type="submit" name="submit"/></p>
+    </form>
+</body>
+</html>
+----
+
+<1> This expression is used to inject a CSRF token into a hidden form field. This token will be verified by the CSRF filter against a CSRF cookie.
+
+You can name the file containing this template as `csrfToken.html` and put it in a `src/main/resources/templates` folder.
+
+Now let's create a resource class which returns an HTML form and handles form POST requests:
+
+[source,java]
+----
+package io.quarkus.it.csrf;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import io.quarkus.qute.Template;
+import io.quarkus.qute.TemplateInstance;
+import io.vertx.ext.web.RoutingContext;
+
+@Path("/service")
+public class UserNameResource {
+
+    @Inject
+    Template csrfToken; <1>
+
+    @Inject
+    RoutingContext routingContext;
```

Comment:
I would very much like to avoid showing injecting the `RoutingContext`. Users should almost never have to do this

Reply:
@geoand Sure, I've added it to show how one can check that the filter has actually been in action to double protect if needed, but I guess just having a note there that a `RoutingContext` attribute is available should be enough - if needed a custom user filter can be added to check it...

Reply:
I just removed this code fragment, users should indeed not be dealing with it to check this property...